packages/ui/v2/core/Meta.tsx

```diff
@@ -0,0 +1,56 @@
+import Head from "next/head";
+import React, { createContext, useContext, useState } from "react";
+
+import { useLocale } from "@calcom/lib/hooks/useLocale";
+
+type MetaType = {
+  title: string;
+  description: string;
+};
+
+const initialMeta = {
+  title: "",
+  description: "",
+};
+
+const MetaContext = createContext({
+  meta: initialMeta,
+  // eslint-disable-next-line @typescript-eslint/no-empty-function
+  setMeta: (newMeta: Partial<MetaType>) => {},
+});
+
+export function useMeta() {
+  return useContext(MetaContext);
+}
+
+export function MetaProvider({ children }: { children: React.ReactNode }) {
+  const [value, setValue] = useState(initialMeta);
+  const setMeta = (newMeta: Partial<MetaType>) => {
+    setValue((v) => ({ ...v, ...newMeta }));
+  };
+
+  return <MetaContext.Provider value={{ meta: value, setMeta }}>{children}</MetaContext.Provider>;
+}
+
+/**
+ * The purpose of this component is to simplify title and description handling.
+ * Similarly to `next/head`'s `Head` component this allow us to update the metadata for a page
+ * from any children, also exposes the metadata via the `useMeta` hook in case we need them
+ * elsewhere (ie. on a Heading, Title, Subtitle, etc.)
+ * @example <Meta title="Password" description="Manage settings for your account passwords" />
+ */
+export default function Meta({ title, description }: MetaType) {
+  const { t } = useLocale();
+  const { setMeta, meta } = useMeta();
+  /* @TODO: maybe find a way to have this data on first render to prevent flicker */
+  if (meta.title !== title || meta.description !== description) {
+    setMeta({ title, description });
+  }
+
+  return (
+    <Head>
+      <title>{t(title)} | Cal.com</title>
+      <meta name="description" content={t(description)} />
```

Comment:
i've noticed that translated descriptions dont work in social previews

we should just use english here, to i18n function @zomars

Reply:
We don't need social previews here. These are being used in the authed screens.

Reply:
Is there a way to make sure these aren't used on public pages?

Reply:
@zlwaterfield not worth the effort to enforce via linting if you ask me. We can add some comments and make sure it doesn't slip on reviews.